### ISSUE TITLE
Authentication failed - invalid request

### ISSUE THREAD

**Issue by Abdull** [microsoftgraph/microsoft-graph-explorer-v4#2250]
**Describe the bug**
After I sign in into my (self-hosted) microsoft-graph-explorer-v4 instance with my organization's account, the Graph Explorer web interface shows
> Authentication failed - invalid request
> Tip - Please sign in again.
![image](https://user-images.githubusercontent.com/529862/203278152-83bbb0f0-d613-4243-a8be-93e177d9e538.png)
* As suggested, signing in again does not help and will only show the same error message.
* After the failed sign-in, running queries still go against the sample account ("Megan Bowen")
* Same error message appears with v7.0.0 and v6.5.0.
* Problem appears in Chromium and Firefox. I have also allowed pop-ups and other browser permissions, but this did not resolve the problem.
* On the other hand, using the [Microsoft-hosted Microsoft Graph Explorer instance](https://developer.microsoft.com/de-DE/graph/graph-explorer), I can successfully sign-in and run queries against my organization's account.

**To Reproduce**
1. Installation:
```
git clone git@github.com:microsoftgraph/microsoft-graph-explorer-v4.git
cd microsoft-graph-explorer-v4.git
git checkout tags/v7.0.0
```
2. Create `.env` with
```
REACT_APP_CLIENT_ID=<"Application (client) ID" from App registrations -> own app -> Overview>
REACT_APP_INSTRUMENTATION_KEY=<"Object ID" from App registrations -> own app -> Overview>
```
3. `npm install`
4. `npm start`
5. Chromium opens, with URL http://localhost:3000 .
6. Graph Explorer appears (successfully allows queries with sample data)
7. Click "Sign In" button
8. A popup appears, allowing the usual https://login.microsoftonline.com/ sign-in flow for my organization's Azure AD login
9. After providing my e-mail address and password, the login popup closes.
10. The Graph Explorer web interface shows the error message

**Expected behavior**
Signing in against my organization's Azure Active Directory shall work with my self-hosted microsoft-graph-explorer-v4 instance.

**Screenshots**
![image](https://user-images.githubusercontent.com/529862/203278152-83bbb0f0-d613-4243-a8be-93e177d9e538.png)

**Desktop (please complete the following information):**
 - Device: Lenovo laptop
 - OS: Debian 11 bullseye
 - Browser: Chromium `107.0.5304.87` ... Firefox `102.4.0esr`

**Comment by Onokaev** [comment]
Hey @Abdull. When registering the application on Azure, did you choose a 'single page application'?
Do you see any errors on the login popup?
Please send screenshots of your app registration process.

**Comment by Abdull** [comment]
So I figured out the root cause of this issue and how to resolve it.

**TL;DR: Open the web interface with `http://localhost:3000/?tenant=$TENANT_ID`, where `$TENANT_ID` is your tenant ID.**

By default, (i.e., when using http://localhost.3000 ), the sign-in process uses the ["multi-tenant" sign-in endpoint](https://learn.microsoft.com/en-us/azure/active-directory/develop/howto-convert-app-to-be-multi-tenant), namely *https://login.microsoftonline.com/common/oauth2/v2.0/authorize* (notice the "*common*" within the URL path).

**But my Azure application is not configured as a multi-tenant/multitenant application; it is configured to only be used by my organization's accounts ("single tenant").** This is configured within https://portal.azure.com under:
*  *Azure Active Directory > App registrations > $OWN_APPLICATION > Authentication > Supported account types*.

![ms-graph-explorer-auth-settings](https://user-images.githubusercontent.com/529862/204764541-6eedd764-58e1-450f-b05c-2dfde54a9d09.png)

IMHO, in general, the "common" sign-in endpoint is not valid/usable for single-tenant applications; single-tenant applications must use the tenant-specific sign-in endpoint, e.g. *https://login.microsoftonline.com/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/oauth2/v2.0/authorize*, where *xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx* is the *Directory (tenant) ID* (`$TENANT_ID`) as reported under
* *Azure Active Directory > App registrations > $OWN_APPLICATION > Overview > Essentials > Directory (tenant) ID*

Looking around the *v7.0.0* source code, I found an **undocumented** way to provide the web application with the tenant ID ( https://github.com/microsoftgraph/microsoft-graph-explorer-v4/blob/v7.0.0/src/modules/authentication/AuthenticationWrapper.ts#L186 ):
* Open the web application with a query/search parameter `tenant=$TENANT_ID`, e.g.:
  * http://localhost:3000/?tenant=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx

Opening the web application this way now allows me to successfully sign in with my organization's account.

I guess it would be good to allow for configuring *microsoft-graph-explorer-v4* with a *tenant ID* in the `.env` file, and use it as the `authority` parameter [inside the *msal-browser* configuration, just as it is already done for the `clientId` there](https://github.com/microsoftgraph/microsoft-graph-explorer-v4/blob/v7.0.0/src/modules/authentication/msal-app.ts#L15) (... this would also require to change `AuthenticationWrapper`, as it seems, with that query parameter checking, to currently overrule any *msal-browser* `authority` configuration).

**Comment by Onokaev** [comment]
This is a very detailed explanation @Abdull. As you've suggested, we can allow for tenant IDs to be supplied on the env file instead of passing it as a query parameter. We can also still allow for both scenarios (tenant ID in the env file or passing it through query params).
Let me discuss this with the team as a feature request. I will close this issue and reference it in a new one as a feature request

**Comment by Abdull** [comment]
Thanks. But I don't understand why this issue is closed then, even as completed.
